From e6e80118e369ad9b572021734ea21d087474fc65 Mon Sep 17 00:00:00 2001
From: Ganapatrao Kulkarni <gkulkarni@caviumnetworks.com>
Date: Thu, 12 May 2016 09:41:21 -0400
Subject: [PATCH 26/39] Revert "arm64: prefetch: enable
 ARM64_HAS_NO_HW_PREFETCH for thunderx 81xx."

This reverts commit 05223364f628fe87f1e07ace80a67c89788297f5.
---
 arch/arm64/include/asm/cputype.h | 2 --
 arch/arm64/kernel/cpufeature.c   | 3 +--
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/arch/arm64/include/asm/cputype.h b/arch/arm64/include/asm/cputype.h
index e3341d0..7540284 100644
--- a/arch/arm64/include/asm/cputype.h
+++ b/arch/arm64/include/asm/cputype.h
@@ -85,12 +85,10 @@
 #define APM_CPU_PART_POTENZA		0x000
 
 #define CAVIUM_CPU_PART_THUNDERX	0x0A1
-#define CAVIUM_CPU_PART_THUNDERX_81XX	0x0A2
 
 #define MIDR_CORTEX_A53 MIDR_CPU_MODEL(ARM_CPU_IMP_ARM, ARM_CPU_PART_CORTEX_A53)
 #define MIDR_CORTEX_A57 MIDR_CPU_MODEL(ARM_CPU_IMP_ARM, ARM_CPU_PART_CORTEX_A57)
 #define MIDR_THUNDERX	MIDR_CPU_MODEL(ARM_CPU_IMP_CAVIUM, CAVIUM_CPU_PART_THUNDERX)
-#define MIDR_THUNDERX_81XX	MIDR_CPU_MODEL(ARM_CPU_IMP_CAVIUM, CAVIUM_CPU_PART_THUNDERX_81XX)
 
 #ifndef __ASSEMBLY__
 
diff --git a/arch/arm64/kernel/cpufeature.c b/arch/arm64/kernel/cpufeature.c
index a62c871..aaf16c2 100644
--- a/arch/arm64/kernel/cpufeature.c
+++ b/arch/arm64/kernel/cpufeature.c
@@ -630,8 +630,7 @@ static bool has_no_hw_prefetch(const struct arm64_cpu_capabilities *entry)
 	rv_min = 0;
 	rv_max = (1 << MIDR_VARIANT_SHIFT) | MIDR_REVISION_MASK;
 
-	return MIDR_IS_CPU_MODEL_RANGE(midr, MIDR_THUNDERX, rv_min, rv_max) ||
-		MIDR_IS_CPU_MODEL_RANGE(midr, MIDR_THUNDERX_81XX, rv_min, rv_max);
+	return MIDR_IS_CPU_MODEL_RANGE(midr, MIDR_THUNDERX, rv_min, rv_max);
 }
 
 static const struct arm64_cpu_capabilities arm64_features[] = {
-- 
1.9.1

